--- ../../../omf-expctl/ruby/omf-expctl/nodeHandler.rb	2013-02-12 10:42:01.493595483 +0100
+++ ../../../../expctl/omf-expctl-5.4/omf-expctl/nodeHandler.rb	2013-02-11 22:24:04.801794495 +0100
@@ -364,20 +364,24 @@
   # mutex and unblock this method.
   #
   def run(main)
+    info "Entered run method!"
     if (@running != nil)
       raise "Already running"
     end
     @running = true
 
+	info "Start profiling"
     Profiler__::start_profile if @doProfiling
 
     begin
       require 'omf-expctl/handlerCommands'
+      info "Load defaultLibs"
       if (@defaultLibs)
         @defaultLibs.split(',').each { |f|
           Experiment.load(f)
         }
       end
+      info "Load extraLibs"
       if (@extraLibs)
         @extraLibs.split(',').each { |f|
           Experiment.load(f)
@@ -389,7 +393,9 @@
     Signal.trap('SIGINT') {Experiment.interrupt}
     # Load the Experiment File , if any
     if @@expFile
+      info "Loading experiment file"
       Experiment.load(@@expFile)
+      info "Starting experiment"
       Experiment.start
     end
 
@@ -563,6 +569,8 @@
     # Parse the command line
     rest = opts.parse(args)
 
+    info "Argumentenenenenne: #{args}"
+
     # Load the Configuration parameters for this EC
     loadControllerConfiguration()
 
@@ -628,11 +636,15 @@
       @@expFile = s
     }
 
+    info "Experiment file::::: #{@@expFile}"
+
     if (@@expFile.nil? && ! (@interactive || @web_ui))
       raise "Missing experiment file"
     end
 
+
     Experiment.expArgs = rest - [@@expFile]
+ 	info "Momenteel hieeeeeerr #{rest}  :::::::::::: #{rest - [@@expFile]}"
   end
 
   #
