--- ../../../omf-expctl/ruby/omf-expctl/oconfig.rb	2013-02-12 10:42:01.494595483 +0100
+++ ../../../../expctl/omf-expctl-5.4/omf-expctl/oconfig.rb	2013-02-11 22:24:03.694794472 +0100
@@ -141,7 +141,7 @@
   # 
   # [Return] the text content of the file, with its mimeType
   #
-  def self.load(uri, evalRuby = false, default_ext = '.rb')
+  def self.load(uri, evalRuby = false, default_ext = '.rb', processor = nil)
     # Find the file to load...
     file = self.findFile(uri)
     if file == nil
@@ -157,8 +157,8 @@
     str = File.new(file).read()
     if evalRuby
       begin
-        eval(str, OMF::EC::CmdContext.instance._binding(),
-             uri)
+	processor = OMF::EC::CmdContext.instance._binding() unless processor
+        eval(str, processor, uri)
       rescue Exception => ex
         if ex.kind_of?(OEDLException)
           # Remove the 1st backtrace line (i.e. the file implementing the cmd)
