--- ../../../omf-expctl/share/repository/system/exp/stdlib.rb	2013-02-12 09:13:50.066378252 +0100
+++ ../../../../expctl/omf-expctl-5.4/repository/system/exp/stdlib.rb	2013-02-11 22:24:06.748794534 +0100
@@ -49,6 +49,11 @@
 #
 everyNS('*', 10) { |ns|
 
+  # ACCOMADATE CONFINE
+  if Experiment.prepare?
+	false
+  else
+
   # First check if the experiment has not been interrupted
   #exp_status = Experiment.state("status/text()")
   #return true if allEqual(exp_status, "INTERRUPTED")
@@ -119,4 +124,5 @@
     # loop and check again in 10sec
     true
   end
+  end
 }
